server/middleware/userMiddlewares.js

```diff
@@ -28,7 +55,17 @@ export const checkEmail = async (req, res, next) => {
 export const parseRegistration = async (req, res, next) => {
   if (Object.keys(req.body).length === 0) {
     res.status(400).json({ status: FAIL, message: resp.emailRequired });
-  } else if (Object.keys(req.body) != "email") {
+  }
+  if (!req.body.email || !req.body.groups) {
+    res.status(400).json({
+      status: FAIL,
+      message: "only email and groups fields are required!"
```

Comment:
Since it is `!req.body.email || !req.body.groups` I suggest:
```suggestion
      message: "Both email and groups fields are required!"
```

Reply:
same case here